db/mongodb.go

```diff
@@ -0,0 +1,192 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"time"
+)
+
+var (
+	cronCollection     = "cronjobs"
+	mongoCollectionMap = map[string]string{
+		"id":       "_id",
+		"IsActive": "active",
+		"Function": "function",
+		"Workload": "workload",
+		"CronTime": "time",
+		"TimeZone": "tz",
+	}
+)
+
+type ErrUnknownField struct {
+	Field string
+}
+
+type ErrUnsupportedWorkloadType struct {
+	Type string
+}
+
+func (e ErrUnknownField) Error() string {
+	return fmt.Sprintf("Unknown cron field: %s", e.Field)
+}
+
+func (e ErrUnsupportedWorkloadType) Error() string {
+	return fmt.Sprintf("Unsupported workload type: %s", e.Type)
+}
+
+type MongoCronJob struct {
```

Comment:
did you consider merging this with `CronJob` by adding the `bson` annotations to it?